app/controllers/agents/invitations_controller.rb

```diff
@@ -1,14 +1,17 @@
 class Agents::InvitationsController < Devise::InvitationsController
-  before_action :set_resource, only: [:update]
 
   def update
-    @agency = self.resource.owned_agency
     super
     if resource.invitation_accepted_at?
-      resource.wholesalers << resource.invited_by.wholesaler
-      resource.agency_applications.update_all(agency_id: @agency.id) if resource.agency_applications.present?
-      UserMailer.welcome(resource.email).deliver_later
-      CompanyOwnerMailer.agent_invite_accept(resource).deliver_later
+      if resource.invited_by.is_a?(Agent)
```

Comment:
Could we do this logic through a concern or inheriting somehow?  Looks strange and feels unorganized that wholesaler logic is in agents controller.

Reply:
i think it could be done with concerns.